crates/matrix-sdk/src/event_handler.rs

```diff
@@ -188,6 +188,7 @@ impl EventHandlerResult for () {
     fn print_error(&self, _event_type: &str) {}
 }
 
+#[cfg(not(any(feature = "anyhow", feature = "eyre")))]
```

Comment:
This means other error types no longer work with either feature activated. That seems like a very bad idea. I'm working on an alternative fix.